### PR TITLE
Add mockgcp Cloud sql v1beta4.

### PR DESCRIPTION
**Pull request by yuwenma**
### Change description

Fixes #

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: mockgcp/mocksql/operations_store.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+func (s *operations) startLRO(ctx context.Context, op *pb.Operation, obj proto.Message, callback func() (proto.Message, error)) (*pb.Operation, error) {
+	log := klog.FromContext(ctx)
+
+	now := time.Now()
+
+	switch obj := obj.(type) {
+	case *pb.DatabaseInstance:
+		op.TargetId = obj.Name
```
Comment (justinsb) [inline mockgcp/mocksql/operations_store.go:43]: Oooh, I like this approach!

File: pkg/webhook/manifests.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 )
 
 var (
-	admissionReviewVersions = []string{"v1"}
+	admissionReviewVersions = []string{"v1", "v1beta1"}
```
Comment (justinsb) [inline pkg/webhook/manifests.go:29]: Is this needed?  It looks accidental...  maybe split into another (non-test) PR if it is needed, or add comments etc

**Comment by justinsb** [comment]
PR lgtm, other than the `admissionReviewVersions` change which looks like a rebase issue?  (I didn't poke around to see where it is used though....)

File: pkg/webhook/manifests.go (hunk 2 above)
Comment (yuwenma) [reply pkg/webhook/manifests.go:29]: Not by accident. The mock test hits the webhook that only allows v1 version. My gut feeling is we may need to further separate some of the mock process from kcc code.  I'll dig deeper next week.

Comment (yuwenma) [reply pkg/webhook/manifests.go]: Fixed. It really took me a while to figure out what happened. In short, [KCC development setup](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/scripts/environment-setup/repo-setup.sh) installs an older kube-apiserver (via kubebuilder v2.3.1) and failed to install a new one (in a later step, the e2e test uses) due to my cloudtop permissions.  The old kube-apiserver only supports v1beta1 but not v1 that breaks the webhook admission review.  What confused me and sent me to the wrong direction is that I'm in a new test env (no cluster, no docker, no gcloud default setup) so I was assuming the dependencies were all up to date.

**Comment by justinsb** [comment]
Thanks @yuwenma  - great stuff!

/approve
/lgtm

**Pull request merged**